odoo/addons/base/models/ir_model.py

```diff
@@ -1389,6 +1389,12 @@ def _process_ondelete(self):
                 selection._get_records().write({field.name: value})
             elif ondelete == 'cascade':
                 selection._get_records().unlink()
+            elif ondelete is None:
+                # special case for custom fields
+                if field.manual and not field.required:
+                    write(selection._get_records(), field.name, False)
+                # nothing to do, the selection does not come from a field extension
+                continue
```

Comment:
Wouldn't it be clearer to let `ondelete` default to `'set null'` above?
```python
ondelete = (field.ondelete or {}).get(selection.value)
# special case for custom fields
if ondelete is None and field.manual and not field.required:
    ondelete = 'set null'

if ondelete is None:
    # nothing to do, the selection does not come from a field extension
    continue
elif callable(ondelete):
    ...
```

Reply:
BTW, if you use my suggestion, you don't need to swap commits.

Reply:
Good catch, the special case for custom fields I discovered after pushing and creating the PR, so I simply added it in there without thinking much about it, this is definitely clearer.

Also originally it was a single commit but I thought splitting them would be best, but I missed the call to `write` indeed